client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -321,6 +333,31 @@ class DeltaSharingRestClient(
     )
   }
 
+  private def getFilesByPage(
+      targetUrl: String,
+      request: QueryTableRequest): (Long, String, Seq[String]) = {
+    val allLines = ArrayBuffer[String]()
+    var updatedRequest = request.copy(maxFiles = Some(maxFilesPerReq))
+    var (version, respondedFormat, lines) = getNDJson(targetUrl, updatedRequest)
+    allLines.appendAll(lines)
+    var nextPageToken = JsonUtils.fromJson[SingleAction](lines.last).nextPageToken
+    while (nextPageToken != null && nextPageToken.token != null && nextPageToken.token.nonEmpty) {
+      // remove last line (nextPageToken) in previous page
```

Comment:
consider adding some logging, and maybe a counter to track the number of pages we get.